File: Proyecto LP/mi-servidor/Frontend/lib/Enums/Motor.dart
enum Motor {
  gasolina("Gasolina", "GASOLINA"),
  diesel("Diesel", "DIESEL"),
  electrico("Electrico", "ELECTRICO"),
  hibrido("Hibrido", "HIBRIDO"),
  gas("Gas", "GAS");

  final String displayName;
  final String backendValue;

  const Motor(this.displayName, this.backendValue);

  static String? getBackendValue(String? displayName) {
    return Motor.values.firstWhere(
      (motor) => motor.displayName == displayName,
      orElse: () => throw Exception("Valor inválido para el enum Motor: $displayName"),
    ).backendValue;
  }
}

File: Proyecto LP/mi-servidor/Backend/src/main/java/com/miapp/mi_servidor/Enums/Motor.java
package com.miapp.mi_servidor.Enums;

public enum Motor {
    GASOLINA("Gasolina"),
    DIESEL("Diesel"),
    ELECTRICO("Electrico"),
    HIBRIDO("Hibrido"),
    GAS("Gas");

    private final String displayName;

    Motor(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }

    public static Motor fromValue(String value) {
        for (Motor motor : Motor.values()) {
            if (motor.getDisplayName().equalsIgnoreCase(value)) {
                return motor;
            }
        }
        throw new IllegalArgumentException("Valor inválido para el enum Motor: " + value);
    }
}
